cloud-controller-manager/do/lb_service_admission_handler.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright 2023 DigitalOcean
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package do
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/digitalocean/godo"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// LBServiceAdmissionHandler validates service type LB.
+type LBServiceAdmissionHandler struct {
+	log        *logr.Logger
+	godoClient *godo.Client
+
+	decoder   *admission.Decoder
+	region    string
+	clusterID string
+	vpcID     string
+}
+
+// NewLBServiceAdmissionHandler returns a configured instance of LBServiceHandler.
+func NewLBServiceAdmissionHandler(log *logr.Logger, godoClient *godo.Client) *LBServiceAdmissionHandler {
+	return &LBServiceAdmissionHandler{
+		log:        log,
+		godoClient: godoClient,
+		decoder:    admission.NewDecoder(runtime.NewScheme()),
+	}
+}
+
+// Handle handles admissions requests for load balancer services.
+func (h *LBServiceAdmissionHandler) Handle(ctx context.Context, req admission.Request) admission.Response {
+	resp := h.handle(ctx, req)
+
+	logFields := []any{"object_name", req.Name, "object_namespace", req.Namespace, "object_kind", req.Kind.String()}
+	if resp.Allowed {
+		h.log.Info("allowing admission request", logFields...)
+	} else {
+		h.log.Info("rejecting admission request", append(logFields, "reason", resp.Result.Message)...)
+	}
+
+	return resp
+}
+
+func (h *LBServiceAdmissionHandler) handle(ctx context.Context, req admission.Request) admission.Response {
+	var svc corev1.Service
+	err := h.decoder.Decode(req, &svc)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, fmt.Errorf("decoding admission request: %w", err))
+	}
+
+	if svc.Spec.Type != corev1.ServiceTypeLoadBalancer {
+		return admission.Allowed("allowing service as it is not a load balancer")
```

Comment:
nit: maybe go with slightly less fancy English here (which I might have suggested in the first place during a Shakespeare-y moment of mine):
```suggestion
		return admission.Allowed("allowing service that is not of type load balancer")
```